crates/bevy_ecs/src/component.rs

```diff
@@ -156,49 +170,76 @@ pub enum StorageType {
     SparseSet,
 }
 
+/// Metadata for a [`Component`] within a [`World`].
+///
+/// [`World`]: crate::world::World
 #[derive(Debug)]
 pub struct ComponentInfo {
     id: ComponentId,
     descriptor: ComponentDescriptor,
 }
 
 impl ComponentInfo {
+    /// Gets the component's unique ID.
     #[inline]
     pub fn id(&self) -> ComponentId {
         self.id
     }
 
+    /// Gets the component's name.
+    ///
+    /// If the component was initialized from a Rust-native type, it's
+    /// name is identical to the output of [`std::any::type_name`].
     #[inline]
     pub fn name(&self) -> &str {
         &self.descriptor.name
     }
 
+    /// Gets the component's type ID, if it has one.
+    ///
+    /// If this returns `None`, there is no matching Rust type for
+    /// the component.
     #[inline]
     pub fn type_id(&self) -> Option<TypeId> {
```

Comment:
This should be `get_type_id()` as it returns an option.

Reply:
I disagree here. The `Option` is not signalling that it failed to fetch, but rather there is no corresponding value at all. On foreign types, it's not failing to fetch a type ID, but that there is none at all.